WordPress/UITests/Tests/AppSettingsTests.swift

```diff
@@ -0,0 +1,33 @@
+import UITestsFoundation
+import XCTest
+
+final class AppSettingsTests: XCTestCase {
+
+    @MainActor
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        setUpTestSuite(removeBeforeLaunching: true)
+
+        try LoginFlow
+            .login(email: WPUITestCredentials.testWPcomUserEmail)
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+        takeScreenshotOfFailedTest()
+    }
+
+    func testImageOptimizationEnabledByDefault() throws {
+        try TabNavComponent().goToMeScreen().goToAppSettings().verifyImageOptimizationSwitch(enabled: true)
+    }
+
+    func testImageOptimizationIsTurnedOnEditor() throws {
+        try TabNavComponent().goToBlockEditorScreen().addImage().chooseOptimizeImages(option: true).closeEditor()
+        try TabNavComponent().goToMeScreen().goToAppSettings().verifyImageOptimizationSwitch(enabled: true)
+    }
+
+    func testImageOptimizationIsTurnedOffEditor() throws {
+        try TabNavComponent().goToBlockEditorScreen().addImage().chooseOptimizeImages(option: false).closeEditor()
+        try TabNavComponent().goToMeScreen().goToAppSettings().verifyImageOptimizationSwitch(enabled: false)
+    }
```

Comment:
Generally, we keep the method chaining start on a new line for each call, to make debugging of separate steps possible:

```suggestion
    func testImageOptimizationEnabledByDefault() throws {
        try TabNavComponent()
            .goToMeScreen()
            .goToAppSettings()
            .verifyImageOptimizationSwitch(enabled: true)
    }

    func testImageOptimizationIsTurnedOnEditor() throws {
        try TabNavComponent()
            .goToBlockEditorScreen()
            .addImage()
            .chooseOptimizeImages(option: true)
            .closeEditor()

        try TabNavComponent()
            .goToMeScreen()
            .goToAppSettings()
            .verifyImageOptimizationSwitch(enabled: true)
    }

    func testImageOptimizationIsTurnedOffEditor() throws {
        try TabNavComponent()
            .goToBlockEditorScreen()
            .addImage()
            .chooseOptimizeImages(option: false)
            .closeEditor()

        try TabNavComponent()
            .goToMeScreen()
            .goToAppSettings()
            .verifyImageOptimizationSwitch(enabled: false)
    }
```

Reply:
Ah good point, thanks for the call out, I applied this suggestion in https://github.com/wordpress-mobile/WordPress-iOS/pull/22095/commits/ac3d4d4e664be4a6c19c2bd06403f65dedbc1ccf.